File: HOT100/HOT100p1/2.双指针/283.移动零.cpp
/*
 * @lc app=leetcode.cn id=283 lang=cpp
 *
 * [283] 移动零
 */

// @lc code=start
class Solution {
    public:
        void moveZeroes(vector<int>& nums) {
            if (nums.size() == 1) return;
            for (int i = 0, j = 1; i < nums.size() && j < nums.size();)
            {
                if (nums[i] == 0 && nums[j] != 0)
                {
                    swap(nums[i], nums[j]);
                }
                else if (nums[i] == 0 && nums[j] == 0)
                {
                    j ++;
                }
                else if (nums[i] != 0 && nums[j] == 0)
                {
                    i ++;
                }
                else
                {
                    i ++, j ++;
                }
            }
            return;
        }
    };
// @lc code=end
class Solution {
    public:
        void moveZeroes(vector<int>& nums) {
            int i = 0, j = 0;
            while (j < nums.size())
            {
                if (nums[j])
                {
                    swap(nums[i], nums[j]);
                    i ++;
                }
                j ++;
            }
        }
    };

File: 题单/&/283.移动零.py
#
# @lc app=leetcode.cn id=283 lang=python3
#
# [283] 移动零
#
from mytools import *
# @lc code=start
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        i = 0
        for j in range(n):
            if nums[j] != 0:
                nums[i] = nums[j]
                i += 1
        for k in range(i, n): nums[k] = 0
# @lc code=end

